docs/scenario-controller.md

```diff
@@ -0,0 +1,43 @@
+<div align="center"><h1>IntegrationTest Scenario Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>Monitor IntegratonTestScenario <br>& filter only created/updated <br>events for the resource ))
+  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureCreatedScenarioIsValid() function
+
+  %% Node definitions
+  
+  application_exists{"Application for scenario <br>was found?"}
+  set_owner_reference(Set owner reference to <br>IntegrationTestScenario <br> if not already existing)
+  environment_defined{"IntegrationTestScenario <br>has environment defined?"}
+  environment_exists{"Environment exists in same <br>namespace as IntegrationTestScenario?"}
```

Comment:
```suggestion
  environment_exists{"Environment exists <br>in same namespace <br>as IntegrationTestScenario?"}
```
breaking this sentence into 3 lines since this node is comparatively bigger than the other decision nodes.